test/solution-template-validation-tests/test/mainTemplateTests.js

```diff
@@ -38,31 +38,28 @@ describe('template files - ', () => {
             mainTemplateFileJSONObject.should.have.property('parameters');
         });
 
-        var parametersInMainTemplate = Object.keys(mainTemplateFileJSONObject.parameters);
+        var currentDir = path.dirname(mainTemplateFile);
+        // assert create ui def exists in the above directory
+        util.assertCreateUiDefExists(currentDir);
 
-        /** Each parameter that does not have a defaultValue MUST have a corresponding output in createUiDefinition.json */
-        it('each parameter that does not have a defaultValue, must have a corresponding output in createUIDef', () => {
-            var currentDir = path.dirname(mainTemplateFile);
-            // assert create ui def exists in the above directory
-            util.assertCreateUiDefExists(currentDir);
+        // get the corresponding create ui def
+        var createUiDefJSONObject = util.getCreateUiDefFile(currentDir).jsonObject;
 
-            // get the corresponding create ui def
-            var createUiDefJSONObject = util.getCreateUiDefFile(currentDir).jsonObject;
+        // get output keys in main template
+        var outputsInCreateUiDef = Object.keys(createUiDefJSONObject.parameters.outputs);
 
-            // get output keys in main template
-            var outputsInCreateUiDef = Object.keys(createUiDefJSONObject.parameters.outputs);
+        // convert to lowercase
+        for (var i in outputsInCreateUiDef) {
+            outputsInCreateUiDef[i] = outputsInCreateUiDef[i].toLowerCase();
+        }
 
-            // convert to lowercase
-            for (var i in outputsInCreateUiDef) {
-                outputsInCreateUiDef[i] = outputsInCreateUiDef[i].toLowerCase();
+        var parametersInMainTemplate = Object.keys(mainTemplateFileJSONObject.parameters);
+        /** Validate each parameter in main template has value in outputs */
```

Comment:
validate each parameter that does not have a defaultValue in mainTemplate, has a value in outputs